File: app/src/main/java/com/dvoroncov/arcore_samples/cloud_anchor/CloudAnchorArFragment.java
package com.dvoroncov.arcore_samples.cloud_anchor;

import com.google.ar.core.Config;
import com.google.ar.core.Session;
import com.google.ar.sceneform.ux.ArFragment;

public class CloudAnchorArFragment extends ArFragment {

    public Session getSession() {
        return getArSceneView().getSession();
    }

    protected Config getSessionConfiguration(Session session) {
        Config config = new Config(session);
        config.setCloudAnchorMode(Config.CloudAnchorMode.ENABLED);
        return config;
    }
}

File: app/src/main/java/com/dvoroncov/arcore/presentation/CloudAnchorArFragment.kt
package com.dvoroncov.arcore.presentation

import com.google.ar.core.Config
import com.google.ar.core.Session
import com.google.ar.sceneform.ux.ArFragment

class CloudAnchorArFragment : ArFragment() {

    val session: Session
        get() = arSceneView.session

    override fun getSessionConfiguration(session: Session): Config {
        val config = Config(session)
        config.cloudAnchorMode = Config.CloudAnchorMode.ENABLED
        return config
    }
}
